src/js/main.js

```diff
@@ -177,21 +203,37 @@ Site.Shapes = {
   timer: null,
   interval: 500,
   animating: false,
+  atLimit: false, // is the map at a limit?
 
   init: function() {
     var _this = this;
 
     _this.showPattern();
 
-    var map = document.getElementById('map');
+    Site.mapElement.addEventListener('startpan', _this.startAnimation.bind(_this));
 
-    map.addEventListener('startpan', function() {
-      _this.startAnimation();
-    });
+    Site.mapElement.addEventListener('stoppan', _this.stopAnimation.bind(_this));
 
-    map.addEventListener('stoppan', function() {
-      _this.stopAnimation();
-    });
+    Site.mapElement.addEventListener('movemap', _this.checkMapPosition.bind(_this));
+
+  },
+
+  // Check if the map position is at a limit
+  checkMapPosition: function(event) {
+    var _this = this;
+
+    var x = event.detail.map.position.x;
+    var y = event.detail.map.position.y;
+
+    // Check for left limit
+    if (x >= 0 ||
+      x <= Site.window.width * -2 || // Check for right limit
+      y >= 0 || // Check for upper limit
+      y <= Site.window.height * -2) { // Check for bottom limit
+      _this.atLimit = true;
+    } else {
```

Comment:
though maybe we should be checking instead only for the corners and the middle of top, right, bottom, and left. where you actually are not panning, but in the pan zone. cause here youre looking for anywhere at the limit. I would say something more like `y == 0 && x <= Site.window.width * -2` for the top-right corner and then `x <= Site.window.width * -2 && y == Site.window.height / 2` for the middle of the right limit, etc? does that make sense? if would end up being 8 conditions (4 for the corners and 4 for the middle of each side), but maybe theres a way to simplify it...
i should have been more explicit in my issue/card. sorry